### PR TITLE
[restangular]: add route-bound methods for scoped service

### PR DESCRIPTION
**Pull request by ngraef**
Documentation: https://github.com/mgonto/restangular#decoupled-restangular-service

When configured via `Restangular.service('route')`, the route is bound to some functions. This change overrides those functions with the correct signatures.

### CODE REVIEW

**Comment by vvakame** [comment]
We can't get response from author. please ping me after 3 days. I'll merge this PR.

File: restangular/restangular.d.ts [hunk 1]
```diff
@@ -90,11 +90,20 @@ declare namespace restangular {
...
-    service(route: string, parent?: any): IService;
+    service(route: string, parent?: any): IScopedService;
     stripRestangular(element: any): any;
     extendModel(route: string, extender: (model: IElement) => any): void;
     extendCollection(route: string, extender: (collection: ICollection) => any): void;
   }
+  
+  interface IScopedService extends IService {
+    one(id: number): IElement;
+    one(id: string): IElement;
```
Comment (andy-ms) [inline restangular/restangular.d.ts]: These could be a single function taking `number|string`.

File: restangular/restangular.d.ts [hunk 2]
```diff
@@ -90,11 +90,20 @@ declare namespace restangular {
...
+    post(elementToPost: any, queryParams?: any, headers?: any): IPromise<any>;
```
Comment (andy-ms) [inline restangular/restangular.d.ts]: Do we need this if we have the second overload?

File: restangular/restangular.d.ts (hunk 1 above)
Comment (ngraef) [inline restangular/restangular.d.ts]: @andy-ms Correct. I simply followed the style of the rest of the document and figured combining union types throughout the file would be a separate PR.

File: restangular/restangular.d.ts (hunk 2 above)
Comment (ngraef) [inline restangular/restangular.d.ts]: @andy-ms Yeah, `post()`'s promise isn't guaranteed to resolve with the same type that's posted. For example, the server might only allow certain properties or settings to be posted, then return the full created object.

Comment (andy-ms) [inline restangular/restangular.d.ts]: The problem is that if someone writes `myScopedService.post(3)` then `IPromise<number>` will not be inferred because the `any` overload comes first. Thus `post<number>` could only be used explicitly. If the promise doesn't actually return the same type as `elementToPost`, then the *second* overload shouldn't exist.

On another note I noticed that `var firstAccount` in the tests is of type `any` and so the typings aren't really getting tested. But that's a job for another PR.

Comment (ngraef) [inline restangular/restangular.d.ts]: Restangular doesn't (and can't) guarantee that the promise returns the same type as `elementToPost`. But a server might make that guarantee, so I suppose the generic overload was added (in b57a5b10723c02624b877cdaf80d0011f54db1a4) as a convenience method for that case in lieu of a type assignment elsewhere.

Regardless, this PR's intention was to fix the behavior of `service` while preserving the semantics of the existing definition. If `post`'s generic overload is going to be removed, that should happen throughout the file, not just here in `IScopedService`.

Comment (andy-ms) [inline restangular/restangular.d.ts]: OK, merging this.
